smoke-test/src/test/java/org/opennms/smoketest/OpenNMSCloudPluginIT.java

```diff
@@ -0,0 +1,163 @@
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.smoketest;
+
+import static java.lang.String.format;
+import static java.util.Objects.nonNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opennms.smoketest.stacks.OpenNMSStack;
+import org.opennms.smoketest.stacks.StackModel;
+import org.opennms.smoketest.utils.KarafShell;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+
+public class OpenNMSCloudPluginIT {
+
+    private static final String pathToFile = "target/plugin-cloud-assembly/org.opennms.plugins.cloud-assembly.kar";
+    private static final String pathToCloudMock = "target/cloud-mock-with-dependencies/org.opennms.plugins.cloud-mock-with-dependencies.jar";
+
+    private KarafShell opennmsShell;
+    private KarafShell sentinelShell;
+
+    @ClassRule
+    public static Network network = Network.newNetwork();
+
+    @ClassRule
+    public static OpenNMSStack stack = OpenNMSStack
+            .withModel(StackModel.newBuilder()
+                    .withSentinel()
+                    .build());
+
+    @ClassRule
+    public static GenericContainer mockCloudContainer = new GenericContainer(DockerImageName.parse("opennms/deploy-base:jre-2.0.6.b165"))
+            .withCreateContainerCmdModifier(cmd -> {
+                final CreateContainerCmd createCmd = (CreateContainerCmd) cmd;
+                createCmd.withEntrypoint("/bin/bash");
+                createCmd.withName("cloudMock");
+            }).withCopyFileToContainer(MountableFile.forHostPath(pathToCloudMock), "/usr/share/opennms/mvn/org.opennms.plugins.cloud-mock-with-dependencies.jar")
+            .withCommand("-c", String.format("java -cp %s %s", "/usr/share/opennms/mvn/org.opennms.plugins.cloud-mock-with-dependencies.jar",
+                    "org.opennms.plugins.cloud.ittest.MockCloudMain")).withExposedPorts(9003)
+            .withNetwork(stack.opennms().getNetwork())
+            .withNetworkAliases("cloudMock");
+
+    @Before
+    public void setUp() {
+        this.opennmsShell = new KarafShell(stack.opennms().getSshAddress());
+        this.sentinelShell = new KarafShell(stack.sentinel().getSshAddress());
+        mockCloudContainer.start();
+    }
+
+    @Test
+    public void install_cloud_plugin_in_opennms_must_be_successful() throws Exception {
+        // Given
+        stack.opennms().copyFileToContainer(MountableFile.forHostPath(pathToFile), "/usr/share/opennms/mvn/org.opennms.plugins.cloud-assembly.kar");
```

Comment:
why don't you copy to /usr/share/opennms/deploy ? 
It is more close to install packages. Also by put in deploy kar:install is not needed.

Reply:
Tried this, but without the kar:install command the test does not seem to work anymore

Reply:
Then the "why" should be examined. Maybe it just needs some `await`ing to ensure the background process has installed the bundles.